### PR TITLE
add support for TinyLlama model

### PR DESCRIPTION
**Pull request by tjs-intel**
:warning: **Do not merge this PR before Habana DeepSpeed 1.15 is released** :warning: 

# What does this PR do?

The [TinyLlama model](https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0) only has checkpoints in the form of `model.safetensors`. This checkpoint needs to be included in the list of checkpoints that is passed to DeepSpeed in order for the model to function properly when initialized with DeepSpeed.

This PR adds the safetensor checkpoints to the list of checkpoints passed to DeepSpeed.

**Note**: This change requires an [upstream commit in microsoft/DeepSpeed](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) to be merged downstream to HabanaAI/DeepSpeed in order for DeepSpeed to support the provided safetensor format.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/habana/transformers/generation/utils.py [hunk 1]
```diff
@@ -542,6 +542,9 @@ def generate(
             generation_config.ignore_eos = kwargs.get("ignore_eos", lazy_mode)
         generation_config.validate()
         model_kwargs = generation_config.update(**kwargs)  # All unused kwargs must be model kwargs
+        if self.config.model_type == "falcon" and "token_type_ids" in kwargs.keys():
+            for key in ["token_type_ids"]:
```
Comment (vidyasiv) [inline optimum/habana/transformers/generation/utils.py:546]: Should this be :
```suggestion
            for key in kwargs["token_type_ids"]:
```

**Comment by regisss** [comment]
@mandy-li Do you know if [this commit](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) will be part of Habana DeepSpeed 1.15?

**Comment by mandy-li** [comment]
> @mandy-li Do you know if [this commit](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) will be part of Habana DeepSpeed 1.15?

@regisss , @tjs-intel , DS changes should be submitted to internal habana deepspeed-fork repo

**Comment by regisss** [comment]
> > @mandy-li Do you know if [this commit](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) will be part of Habana DeepSpeed 1.15?
> 
> @regisss , @tjs-intel , DS changes should be submitted to internal habana deepspeed-fork repo

https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289 is part of the v0.13 release of DeepSpeed. I guess it should be part of Habana DeepSpeed 1.15 or 1.16 no?

**Comment by mandy-li** [comment]
> > > @mandy-li Do you know if [this commit](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) will be part of Habana DeepSpeed 1.15?
> > 
> > 
> > @regisss , @tjs-intel , DS changes should be submitted to internal habana deepspeed-fork repo
> 
> [microsoft/DeepSpeed@c8c57b8](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) is part of the v0.13 release of DeepSpeed. I guess it should be part of Habana DeepSpeed 1.15 or 1.16 no?

Yes, will be in 1.15, but we modified it in the internal Habana DS repo. 
@tjs-intel , this PR should go to habana oh-fork to test internally.

**Comment by tjs-intel** [comment]
OH-fork PR here https://github.com/HabanaAI/optimum-habana-fork/pull/25

File: optimum/habana/transformers/generation/utils.py (hunk 1 above)
Comment (tjs-intel) [reply optimum/habana/transformers/generation/utils.py]: I based my changes off https://github.com/HabanaAI/optimum-habana-fork/pull/15 so I think this code is meant to support falcon-180b inference. It is not relevant for the change I'd like to make.

Comment (mandy-li) [reply optimum/habana/transformers/generation/utils.py]: @tjs-intel , for any falcon-180b inference related code change, please work with @schoi-habana to submit PR to internal oh-fork repo.

Comment (tjs-intel) [reply optimum/habana/transformers/generation/utils.py]: I have done the opposite and removed the code change related to falcon-180b

**Comment by tjs-intel** [comment]
https://github.com/HabanaAI/optimum-habana-fork/pull/25 has been merged

**Comment by regisss** [comment]
> [HabanaAI#25](https://github.com/HabanaAI/optimum-habana-fork/pull/25) has been merged

So we need to wait for 1.15 to be merged to merge this PR right?

**Comment by tjs-intel** [comment]
@regisss I will leave that discussion to the maintainers.

The consequence of merging this before the HabanaAI/DeepSpeed 1.15 release is that the latest released version of HabanaAI/DeepSpeed (1.14) does not support safetensors. If there are any models with both `.bin` and `.safetensors` checkpoints, this change will prefer the `.safetensors` checkpoint and send that to DeepSpeed instead of the `.bin` checkpoint. This will cause DeepSpeed to raise an exception since the `.safetensors` checkpoint cannot be unpickled.

**Review by regisss — COMMENTED** [review]
Thanks for the clear explanation @tjs-intel! In that case, we'll wait for the release of SynapseAI v1.15 to merge this PR.
Changes look good to me, could you also run the following to make the code style check pass please?
```
pip install -U ruff
make style
```

**Comment by regisss** [comment]
Merging in the `synapse_1.15` branch

**Pull request merged**